sources/api/netdog/src/cli/mod.rs

```diff
@@ -165,6 +167,50 @@ where
     Ok(())
 }
 
+// Gather net config from possible sources, returning both the config and the source
+fn fetch_net_config() -> Result<(Option<Box<dyn Interfaces>>, PathBuf)> {
+    let over_ride = Path::new(OVERRIDE_NET_CONFIG_FILE);
+    let default = Path::new(DEFAULT_NET_CONFIG_FILE);
+    let cmdline = Path::new(KERNEL_CMDLINE);
```

Comment:
nit: this might eliminate the need for `to_path_buf()` calls later:
```suggestion
    let default = PathBuf::from(DEFAULT_NET_CONFIG_FILE);
    let cmdline = PathBuf::from(KERNEL_CMDLINE);
```